src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetService.cs

```diff
@@ -108,4 +133,66 @@ private static DataSetViewModel MapDataSet(DataSet dataSet)
             .AsNoTracking()
             .FirstOrNotFoundAsync(p => p.Id == publicationId, cancellationToken: cancellationToken);
     }
+
+    private async Task<Either<ActionResult, ReleaseVersion>> CheckReleaseVersionExists(
+        Guid releaseVersionId,
+        CancellationToken cancellationToken = default)
+    {
+        return await contentDbContext.ReleaseVersions
+            .AsNoTracking()
+            .SingleOrNotFoundAsync(rv => rv.Id == releaseVersionId, cancellationToken: cancellationToken);
+    }
+
+    private async Task<Either<ActionResult, DataSetVersion>> CheckVersionExists(
+        Guid dataSetId,
+        string dataSetVersion,
+        CancellationToken cancellationToken = default)
+    {
+        if (!VersionUtils.TryParse(dataSetVersion, out var version))
+        {
+            return new NotFoundResult();
+        }
+
+        return await publicDataDbContext.DataSetVersions
+            .AsNoTracking()
+            .Include(dsv => dsv.DataSet)
+            .Where(dsv => dsv.DataSetId == dataSetId)
+            .Where(dsv => dsv.VersionMajor == version.Major)
+            .Where(dsv => dsv.VersionMinor == version.Minor)
+            .SingleOrNotFoundAsync(cancellationToken);
+    }
+
+    private async Task<Either<ActionResult, Unit>> CheckDataSetVersionBelongsToReleaseVersion(
+        DataSetVersion dataSetVersion,
+        ReleaseVersion releaseVersion)
+    {
+        return await contentDbContext.ReleaseFiles
+            .AsNoTracking()
+            .Where(rf => rf.ReleaseVersionId == releaseVersion.Id)
+            .Where(rf => rf.FileId == dataSetVersion.CsvFileId)
+            .SingleOrNotFoundAsync()
+            .OnSuccessVoid();
```

Comment:
I actually personally prefer keeping them all on separate lines as I think it's more readable. Also becomes unwieldy if there are a lot of conditions! So I prefer to stick to the convention of 1 per line. Wonder if we should discuss this as a potential ADR?

Reply:
Thanks, I'm happy for you to keep it as it is.

I was just pointing out that it can be done without `Where` by using the predicate parameter of `SingleOrNotFoundAsync` instead. The condition could still be formatted over separate lines.

I actually like using multiple `Where`'s too in other cases. 🙂.

It does sound like something we could discuss with the rest of the team, yeah.